##Урок 8. Корреляционный анализ

1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.


In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np

In [ ]:
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

$cov_{zpks}=M(zpks)-M(zp)*M(ks)$

In [ ]:
zpks=list(map(lambda x,y: x*y, zp, ks))
zpks

[14035, 25830, 166060, 183800, 18360, 51730, 35262, 135300, 89520, 91520]

In [ ]:
def mean_(data:list):
    return sum(data)/10

cov = mean_(zpks) - mean_(zp) * mean_(ks)
cov

9157.839999999997

In [ ]:
np.cov(zp, ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

Коэффициент корреляции Пирсона:  
$r_{x,y}=\frac{cov_{x,y}}{\sigma_x\cdot\sigma_y}$

In [ ]:
sigma_zp = np.std(zp, ddof=0)
sigma_ks = np.std(ks, ddof=0)
r=cov/(sigma_zp*sigma_ks)
r

0.8874900920739158

In [ ]:
np.corrcoef(zp,ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах: 131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.


Поскольку в задаче имеется выборка малого объема, применим распределение Стьюдента.$$ \bar X \pm t_{\alpha/2}\cdot \frac {\sigma}{\sqrt n}$$
n=10

In [ ]:
import scipy.stats as stats
import numpy as np

data=[131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
x = np.mean(data)
sigma = np.std(data, ddof=1)
print(f'Выборочное среднее-{x}\nСреднее квадратическое отклонение-{sigma}')

Выборочное среднее-118.1
Среднее квадратическое отклонение-10.54566788359614


$\bar X=118.1$  
$\sigma = 10.55$

Найдем t-критерий, табличный

In [ ]:
t=stats.t.ppf(0.975,len(data)-1)
t

2.2621571627409915

$$ 118.1 \pm 2.26\cdot \frac {10.55}{\sqrt 10}$$

In [ ]:
import math
print(118.1+2.226*(10.55/math.sqrt(10)))
118.1-2.226*(10.55/math.sqrt(10))

125.52638772546922


110.67361227453077

**Ответ**: [110.67; 125.53]

3. Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

Поскольку известна дисперсия генеральной совокупности, то
используем формулу с z критерием:
$$ \bar X \pm z_{\alpha/2}\cdot \frac {\sigma}{\sqrt n}$$

$\bar X = 174.2$  
z от 2.5% = 1.96 по таблице  
$\sigma=\sqrt{25}=5$  
$n=27$

In [ ]:
import math
print(174.2+1.96*(5/math.sqrt(27)))
174.2-1.96*(5/math.sqrt(27))

176.08601087935276


172.31398912064722

**Ответ**: [172.31; 176.08]